modules/lang-expression/src/main/java/org/elasticsearch/script/expression/ExpressionScriptEngine.java

```diff
@@ -453,13 +453,13 @@ private static DoubleValuesSource getDocValueSource(String variable, SearchLooku
         }
 
         String fieldname = parts[1].text;
-        MappedFieldType fieldType = lookup.doc().fieldType(fieldname);
+        MappedFieldType fieldType = lookup.fieldType(fieldname);
 
         if (fieldType == null) {
             throw new ParseException("Field [" + fieldname + "] does not exist in mappings", 5);
         }
 
-        IndexFieldData<?> fieldData = lookup.doc().getForField(fieldType);
+        IndexFieldData<?> fieldData = lookup.getForField(fieldType);
```

Comment:
it is odd that we need two additional methods in SearchLookup. Here we could have a single getForField(String) but I guess in other places that would not work?

Reply:
We may be able to remove them entirely, let me try!

Reply:
Alas, we need both here - we need access to both the field data and the MappedFieldType